core/src/calleeFunctions/index.ts

```diff
@@ -18,30 +25,95 @@ const allCalleeFunctions: Record<CalleeNames, CalleeFunctions> = {
     UniswapV3Callee,
 };
 
-export const getCalleeData = async function (
+export const getCalleesData = async function (
     network: string,
     collateralType: string,
     profitAddress: string
-): Promise<string> {
+): Promise<Record<string, string>> {
     const collateral = getCollateralConfigByType(collateralType);
-    const calleeFuctions = allCalleeFunctions[collateral.exchange.callee];
-    if (!calleeFuctions) {
+    const calleeFuctions = {} as Record<string, CalleeFunctions>;
+    Object.entries(collateral.exchanges).forEach(([key, value]) => {
+        calleeFuctions[key] = allCalleeFunctions[value.callee];
+    });
+    if (!Object.keys(calleeFuctions).length) {
         throw new Error(`Unsupported collateral type "${collateralType}"`);
     }
-    return await calleeFuctions.getCalleeData(network, collateral, profitAddress);
+    const calleesData = {} as Record<string, string>;
+    Object.entries(calleeFuctions).forEach(async ([key, value]) => {
+        calleesData[key] = await value.getCalleeData(network, collateral, profitAddress);
+    });
+    return calleesData;
 };
 
-const _getMarketPrice = async function (
+export const getMarketData = async function (
     network: string,
     collateralSymbol: string,
     amount: BigNumber = new BigNumber('1')
-): Promise<BigNumber> {
+): Promise<Record<string, MarketData>> {
     const collateral = getCollateralConfigBySymbol(collateralSymbol);
-    const calleeFuctions = allCalleeFunctions[collateral.exchange.callee];
-    if (!calleeFuctions) {
+    const calleeIdAndFuctions = {} as Record<string, CalleeFunctions>;
+    Object.keys(collateral.exchanges).forEach(id => {
+        calleeIdAndFuctions[id] = allCalleeFunctions[collateral.exchanges[id].callee];
+    });
+    if (!Object.keys(calleeIdAndFuctions).length) {
         throw new Error(`Unsupported collateral symbol "${collateralSymbol}"`);
     }
-    return await calleeFuctions.getMarketPrice(network, collateral, amount);
+    const marketData = {} as Record<string, MarketData>;
+    const gasPrice = await getGasPriceForUI(network);
+    const exchangeFee = gasPrice.multipliedBy(722651 - 145438);
```

Comment:
Yes, but
- this will calculate `exchangeFee` in ETH, you will need to convert it into DAI
- please move this calculation in to `fees` file
- please put numbers into variables eg `SWAP_TRANSACTION_GAS_LIMIT - BID_TRANSACTION_GAS_LIMIT` that are reused in other places in the same file too